### DISCUSSION TITLE
Some thoughts on fresh

### DISCUSSION THREAD

**Discussion by posporus** [freshframework/fresh#187] [Ideas]
Hello fresh people,

I am currently exploring and learning to appreciate Deno. In the process, I of course stumbled across fresh. I really like the approach, which is inspired by Next without a build-step. But instead of the build-step, a manifest has to be created with a CLI. I have some thoughts on this and would like to know if I have understood everything correctly.

Why is there a CLI at all? The only reason I can see is to create a manifest. Couldn't this step be skipped, for example by using a dev-command in the start script, which then starts a watcher, which monitors the filesystem and then restarts the server or even just reloads single resources? At the same time, this would solve the problem that new routes and islands are not automatically detected.
The second task of the CLI - initiating a new project - could easily be replaced with a git-clone for example. Or am I misunderstanding the task of the CLI or the manifest? Can someone enlighten me?

Either way - thanks for the great work and for bringing forward such good and important projects!

**Comment by kuboon** [comment]
Deployed production instance does not use file watching and manifest updating. CLI is only for local development and preparing deploy image.

**Comment by sunw31** [comment]
Absolutely, I agree with you, maybe "fresh“ is deno-deploy version nextjs.

Also I suggest fresh should integrete with @trpc , like blitzjs design.

**Reply by lantos1618** [reply]
Second this, having types across the wire is probably one of the best winners for a feature.

There is trpc and gentleRPC that could be potential solutions but having the feature as a part of the fresh frame work would be a big buy in. Teams could (mostly) sidestep openAPI/Swagger/Graphql complexities. 

I have see it as a todo long term feature in the roadmap but probably something that would be cool to bring forward.

Theo also mentioned this in a video with ~55K views.

snipped to his thoughts.
https://youtu.be/1pcBlSGW_Sk?t=1589

Some viewers in the comments tried to make a TRCP stack with fresh.

Quite excited about this project will keep eyes on.

**Reply by bjesuiter** [reply]
I finished my blogpost today about how to integrate tRPC with Fresh: 

https://blog.codemonument.com/2022-11-03-trpc-in-fresh-islands

**Comment by lucacasonato** [comment]
Hey, thanks for the feedback!

In the limit I want to remove the `fresh` CLI. It is a stop-gap solution to work around a few shortcomings in Deno right now. I do expect there to always be a Deno script that one can run to set up a project (imagine `deno run -A https://fresh.deno.dev/setup.ts`). This has many benefits over a Git "template" repo:

- I can easily keep an in tree setup script in sync with the fresh source code. This is not as easy if it is in a seperate repo.
- Once the setup becomes parametric (eg "Do you want to automatically add `twind` styling support? (y/n)"), you need a new repo for each combination of setup options. A startup script is much more dynamic.

The `fresh manifest` command will soon be gone: #205
